internal/util/retrypolicy/retry.go

```diff
@@ -0,0 +1,51 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package retrypolicy
+
+import (
+	"context"
+	"log/slog"
+	"time"
+
+	"github.com/cartesi/rollups-node/internal/util/reflection"
+)
+
+// Implements a simple method call retry policy.
+// This retry policy will retry a failed execution for up to 'maxRetries' times
+// In between retry calls it will wait a "maxDelay"
+func CallFunctionWithRetryPolicy[
+	R any,
+	A any,
+](
+	fn func(A) (R, error),
+	args A,
+	maxRetries uint,
+	maxDelay time.Duration,
+) (R, error) {
+
+	var lastErr error
+	var lastValue R
+	var funcName string
+	if slog.Default().Enabled(context.Background(), slog.LevelDebug) {
+		funcName = reflection.GetFunctionName(fn)
+	}
+	for i := uint(0); i <= maxRetries; i++ {
+		lastValue, lastErr = fn(args)
+		if lastErr == nil {
+			return lastValue, nil
+		}
+		if slog.Default().Enabled(context.Background(), slog.LevelDebug) {
```

Comment:
Retrying is not debug, should be info level. Please, also log the wait duration.